### PR TITLE
Selectors: allow for computed selectors in the target selector portion of an array like selector.

### PR DESCRIPTION
**Pull request by mikesherov**

### CODE REVIEW

File: test/behavior/dynamic.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ define(['index'], function(Focss) {
     return window.getComputedStyle(el[0] || el)[prop];
   }
 
-  describe('dynamic rules', function() {
+  fdescribe('dynamic rules', function() {
```
Comment (jrefano) [inline test/behavior/dynamic.js]: rm f

**Comment by jrefano** [comment]
LGTM

**Comment by Attamusc** [comment]
:+1:

File: src/Engine.js [hunk 2]
```diff
@@ -147,15 +148,30 @@ define([
       }, this);
     },
 
+    _getArtifactsFromSelector: function(selector) {
+      var artifacts = {},
+          expr;
+
+      while ((expr = this.constructor.Rule.computed.exec(selector)) !== null) {
```
Comment (mrjoelkemp) [inline src/Engine.js]: I'm a little confused here. You're neither modifying `expr` nor `selector` within the loop, so how does this terminate?

Comment (mikesherov) [inline src/Engine.js]: the way exec works when you use `//g`.

**Pull request merged**